Patch-Source: https://github.com/AOSC-Dev/aosc-os-abbs/blob/a2b3abf61f9244342b88a2f1dd19e9fada1825bd/app-web/firefox/autobuild/patches/0005-AOSCOS-Guard-stray-inclusion-of-Simulator-riscv64.h.patch
From 655977ebef40cd99e2fd6f9f1a79693666e98bd1 Mon Sep 17 00:00:00 2001
From: Rong Bao <webmaster@csmantle.top>
Date: Thu, 18 Jun 2026 16:43:16 +0800
Subject: [PATCH 5/5] AOSCOS: Guard stray inclusion of Simulator-riscv64.h

This is a leftover by Bug 2038968 D300034, but it should be fixed in Bug
2039343 D300387 which happens to miss the Fx152 train.

Should be dropped by the release of Fx153.

Link: https://phabricator.services.mozilla.com/D300034 ("Bug 2038968 -
 Part 4: Make it an error to include simulator headers when simulator is
 disabled. r=jandem!")
Link: https://phabricator.services.mozilla.com/D300387 ("Bug 2039343 -
 Part 6: Remove InstructionGetters. r=jandem!")
---
 js/src/jit/riscv64/constant/Base-constant-riscv.cpp | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/js/src/jit/riscv64/constant/Base-constant-riscv.cpp b/js/src/jit/riscv64/constant/Base-constant-riscv.cpp
index b25ce174bc5bbb3a01ebd7676c91afa7b3d5a628..40b7472ccec197c88de7c7a66809b8822038dc6c 100644
--- a/js/src/jit/riscv64/constant/Base-constant-riscv.cpp
+++ b/js/src/jit/riscv64/constant/Base-constant-riscv.cpp
@@ -13,7 +13,11 @@
 #include "jit/riscv64/constant/Constant-riscv-v.h"
 #include "jit/riscv64/constant/Constant-riscv-zicsr.h"
 #include "jit/riscv64/constant/Constant-riscv-zifencei.h"
-#include "jit/riscv64/Simulator-riscv64.h"
+
+#ifdef JS_SIMULATOR_RISCV64
+#  include "jit/riscv64/Simulator-riscv64.h"
+#endif
+
 namespace js {
 namespace jit {
 
-- 
2.54.0

